### PR TITLE
feat: rename logit_bias/logit_scale to logit_mean/logit_sigma for affine score calibration

### PR DESCRIPTION
**Pull request by jefp**
## Purpose

Rename the affine score calibration parameters in `PoolerConfig` to align with LayerNorm convention, as suggested by @noooop in #39435:

- `logit_bias` → `logit_mean` (same behavior: subtract from logits)
- `logit_scale` → `logit_sigma` (divide instead of multiply)

Formula: `activation((logit - logit_mean) / logit_sigma)`

The old parameters are kept as deprecated aliases with automatic conversion. `logit_scale` is auto-converted via `logit_sigma = 1/logit_scale`.

Follow-up to #39435 (merged).

## Changes

- `vllm/config/pooler.py` — Add `logit_mean`/`logit_sigma` fields, deprecation handling in `__post_init__`, zero-division guards
- `vllm/model_executor/layers/pooler/seqwise/heads.py` — Rename to `logit_mean`/`logit_sigma`, change `*=` to `/=`
- `vllm/model_executor/layers/pooler/seqwise/poolers.py` — Read from new config fields
- `vllm/model_executor/layers/pooler/tokwise/heads.py` — Same as seqwise
- `vllm/model_executor/layers/pooler/tokwise/poolers.py` — Same as seqwise
- `vllm/model_executor/models/config.py` — Update JinaVL to use `logit_mean`
- `docs/models/pooling_models/classify.md` — Updated docs with new parameter names

**7 files changed, ~85 lines. Fully backward compatible.**

## Backward compatibility

Old syntax still works with deprecation warnings:
```bash
# Deprecated (still works, logs warning)
--pooler-config '{"use_activation": true, "logit_bias": 4.5, "logit_scale": 0.5}'

# New (recommended)
--pooler-config '{"use_activation": true, "logit_mean": 4.5, "logit_sigma": 2.0}'
```

When both old and new are set, raises `ValueError` to prevent conflicts.

## Validation

- `logit_scale=0` raises `ValueError` (division by zero in conversion)
- `logit_sigma=0` raises `ValueError` (division by zero in forward pass)

## Test plan

- [x] Ruff format + check passes
- [x] mypy 3.10 passes
- [x] Tested with reranker model — new and deprecated syntax produce identical scores
- [x] Deprecation warnings display correctly
- [x] JinaVL config updated, existing models unaffected

**Not a duplicate:** This is a follow-up rename requested by @noooop in the review of #39435.

AI-assisted work: AI assistance was used for implementation. All code reviewed and tested by human submitter.

Cc: @noooop @DarkLight1337

### CODE REVIEW

File: vllm/model_executor/layers/pooler/seqwise/heads.py [hunk 1]
```diff
@@ -140,10 +140,11 @@ def forward(
...
         # logits shape: [batchsize, num_labels]
-        if self.logit_bias is not None:
-            logits -= self.logit_bias
-        if self.logit_scale is not None:
-            logits *= self.logit_scale
+        # Affine score calibration: activation((logit - mean) / sigma)
+        if self.logit_mean is not None:
+            logits -= self.logit_mean
+        if self.logit_sigma is not None:
+            logits /= self.logit_sigma
```
[the remark this answers is not in the document]
Comment (jefp) [reply vllm/model_executor/layers/pooler/seqwise/heads.py]: Good point. Changed to out-of-place operations.

[the remark this answers is not in the document]
Comment (jefp) [reply vllm/model_executor/layers/pooler/seqwise/heads.py]: Good point. Changed to out-of-place operations.

File: vllm/model_executor/layers/pooler/tokwise/heads.py [hunk 2]
```diff
@@ -127,10 +127,11 @@ def forward_chunk(
...
 
-        if self.logit_bias is not None:
-            logits -= self.logit_bias
-        if self.logit_scale is not None:
-            logits *= self.logit_scale
+        # Affine score calibration: activation((logit - mean) / sigma)
+        if self.logit_mean is not None:
+            logits -= self.logit_mean
+        if self.logit_sigma is not None:
+            logits /= self.logit_sigma
```
[the remark this answers is not in the document]
Comment (jefp) [reply vllm/model_executor/layers/pooler/tokwise/heads.py]: Done. Same fix applied here.

**Review by noooop — APPROVED** [review]
Thanks for your contribution.

File: vllm/config/pooler.py [hunk 3]
```diff
@@ -105,6 +119,39 @@ class PoolerConfig:
...
     def __post_init__(self) -> None:
+        # Handle deprecated logit_bias → logit_mean
+        if self.logit_bias is not None:
+            if self.logit_mean is not None:
+                raise ValueError(
+                    "Cannot set both `logit_bias` and `logit_mean`. "
+                    "`logit_bias` is deprecated, use `logit_mean` instead."
+                )
+            logger.warning(
+                "`logit_bias` is deprecated and will be removed in a future release. "
```
Comment (DarkLight1337) [inline vllm/config/pooler.py]: Can you set v0.21?

**Review by DarkLight1337 — APPROVED** [review]
